whois/web.py

```diff
@@ -275,9 +273,15 @@ def login():
         except User.DoesNotExist:
             user = None
 
-        if user is not None and user.auth(request.form["password"]) is True:
-            login_user(user)
-            app.logger.info("logged in: {}".format(user.username))
+        if user is not None:
+
+            if user.auth(request.form["password"]) is True:
+                login_user(user)
+                app.logger.info("logged in: {}".format(user.username))
+            else:
```

Comment:
```suggestion
            elif user.is_sso: 
```
Not every failed login should lead to redirect. 
Consider also adding this:
```
else:
    pass
```

Reply:
fixed